### ISSUE TITLE
optimize extruder movement to exclude unnecessary moves across the layer (extruder on the end of one element starts with nearest element)

### ISSUE THREAD

**Issue by woodyv** [slic3r/Slic3r#4747]
### Do not request features for the Prusa3D fork of Slic3r here without confirming it is applicable to the mainline Slic3r, or your issue will be closed. [Development builds of Slic3r](https://dl.slic3r.org/dev). [Prusa3D Issue Tracker](https://github.com/Prusa3D/Slic3r/issues)

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by supermerill** [comment]
https://en.wikipedia.org/wiki/Travelling_salesman_problem

**Comment by woodyv** [comment]
yes I know, but wave algoritm (as 4example in dps placement and routing) is good enough. in a simplest way it is posible to sort g commands in nearest end/begin point before output to the file.
